src/coinbase/webhook.ts

```diff
@@ -0,0 +1,169 @@
+import { Webhook as WebhookModel, WebhookEventType, WebhookEventFilter } from "../client/api";
+import { Coinbase } from "./coinbase";
+import { InternalError } from "./errors";
+
+/**
+ * A representation of a Webhook,
+ * which provides methods to create, list, update, and delete webhooks that are used to receive notifications of specific events.
+ */
+export class Webhook {
+  private model: WebhookModel | null;
+
+  /**
+   * Initializes a new Webhook object.
+   *
+   * @param model - The underlying Webhook object.
+   * @throws {InternalError} If the model is not provided.
+   */
+  constructor(model: WebhookModel) {
+    if (!model) {
+      throw new InternalError("Webhook model cannot be empty");
+    }
+    this.model = model;
+  }
+
+  /**
+   * Creates a new webhook for a specified network.
+   *
+   * @param networkId - The network ID for which the webhook is created.
+   * @param notification_uri - The URI where notifications should be sent.
+   * @param event_type - The type of event for the webhook.
+   * @param event_filters - Filters applied to the events that determine which specific events trigger the webhook.
+   * @returns A promise that resolves to a new instance of Webhook.
+   */
+  public static async create(
+    networkId,
+    notification_uri,
+    event_type,
+    event_filters,
+  ): Promise<Webhook> {
+    const result = await Coinbase.apiClients.webhook!.createWebhook({
```

Comment:
how is alert created?

Reply:
i meant how is this surfaced to user? would `result.data` be empty when we do `new Webhook(result.data)`?

Reply:
it could be empty, then from user would see `InternalError("Webhook model cannot be empty");` ; check the constructor code couple lines above